src/token.ts

```diff
@@ -0,0 +1,16 @@
+/**
```

Comment:
nit: I think this interface should go into `exchange.ts` since that is where it is used (and created). Note that interfaces in TypeScript are duck-typed. That means the following works.
```ts
interface Foo {
    a: string,
    b: number,
}
interface Bar {
    b: number,
    c?: boolean,
}
function takesBar(bar: Bar) {
    console.log(`${bar.b} ${bar.c ?? true}`)
}

let foo: Foo = { a: "answer", b: 42 };
takesBar(foo)
```